### ISSUE TITLE
expose the navigator route as a service extension

### ISSUE THREAD

**Issue by devoncarew** [flutter/flutter#3415]
Something like:

- [ ] get the current value for the route
- [ ] set a new value for the route
- [ ] enable and disable receiving events on route changes

@Hixie, I very briefly looked at the flutter sources, but didn't see how to get this info.

**Comment by eseidelGoogle** [comment]
This would be if IntelliJ wants to display it.

**Comment by devoncarew** [comment]
Yes, we would like to display this. When we discussed this previously the main issue that was brought up was that there was no concept of a single navigator.

**Comment by Hixie** [comment]
On cupertino (and indeed in any app, but it's common for cupertino apps) there can be multiple navigators. How would you want to handle that?

**Comment by Piinks** [comment]
I am going to close this as superseded by the development of Dart DevTools.
The modern Widget Inspector allows for deep inspection of the widget tree, including the state of any Navigator in the application. Additionally, as noted in the issue, the presence of multiple Navigators makes a single service extension for "the" route problematic, whereas the Inspector handles this naturally.

If there is more work to do here, let us know by filing a new issue with up to date information. Thanks! :)
